core/client/src/main/java/alluxio/client/block/BlockStoreContext.java

```diff
@@ -313,11 +227,50 @@ public static void releaseNettyChannel(InetSocketAddress address, Channel channe
   }
 
   /**
+   * Acquires a block worker thrift client from the block worker thrift client pools.
+   *
+   * @param address the address of the block worker
+   * @return the block worker thrift client
+   * @throws IOException if it fails to connect to remote worker
+   */
+  public static BlockWorkerClientService.Client acquireBlockWorkerThriftClient(
+      final InetSocketAddress address) throws IOException {
+    if (!BLOCK_WORKER_THRIFT_CLIENT_POOL.containsKey(address)) {
+      BlockWorkerThriftClientPool pool = new BlockWorkerThriftClientPool(address,
+          Configuration.getInt(PropertyKey.USER_BLOCK_WORKER_CLIENT_POOL_SIZE_MAX),
+          Configuration.getInt(PropertyKey.USER_BLOCK_WORKER_CLIENT_POOL_GC_THRESHOLD_SECS));
+      if (BLOCK_WORKER_THRIFT_CLIENT_POOL.putIfAbsent(address, pool) != null) {
+        pool.close();
+      }
+    }
+    try {
+      return BLOCK_WORKER_THRIFT_CLIENT_POOL.get(address).acquire();
```

Comment:
yeah, that is on purpose. Any concern? Note that if the hashmap operation (get) here becomes expensive, it is easy to to cache the pool the client (e.g. RetryHandlingBlockWorkerClient). 

And the size of the hashmap is limited by the number of Alluxio workers which is unlikely to cause problem.